numpy/random/_generator.pyx

```diff
@@ -4414,7 +4415,15 @@ cdef class Generator:
                 with self.lock, nogil:
                     _shuffle_raw_wrap(&self._bitgen, n, 1, itemsize, stride,
                                       x_ptr, buf_ptr)
-        elif isinstance(x, np.ndarray) and x.ndim and x.size:
+        elif isinstance(x, np.ndarray):
+            if axis >= x.ndim:
```

Comment:
This doesn't throw the exception for `axis=-12`, does it? I think it would be better to just let `swapaxes` handle the exception - it doesn't really coat anything to call that on empty arrays, and it means we don't repeat the error handling logic.

Reply:
Actually I take that back, this codepath is already unreachable, the axis is normalized in the first line of this function.